_posts/2020-06-07-Human-Curation-and-Digital-Datasets.md

```diff
@@ -69,64 +70,160 @@ source to check on possible transcription errors originating from the
+<#Flaithbertach>
+a foaf:Person;
+irishRel:nomName "Flaithbertach";
+rel:childOf <#Crunmael>.
+
+<#Crunmael>
+a foaf:Person;
+irishRel:genName "Crunmael";
+rel:childOf <#Commáin>.
+
+<#Commáin>
+a foaf:Person;
+irishRel:genName "Commáin";
+rel:childOf <#Fhinain>.
+
+<#Fhinain>
+a foaf:Person;
+irishRel:genName "Fhinain";
+rel:childOf <#Faigfhir>.
+
+<#Faigfhir>
+a foaf:Person;
+irishRel:genName "Faigfhir";
+rel:childOf <#Ernine>.
+...
+```
+
+One thing to note here is that the utilities may evolve and change
+over time to become more sophisticated depending or even be deleted
+entirely based on the interests and needs of those involved.
+
+The way in which this works is to split the String Pedigree by the "m"
+or other textual means of indicating "son of". The list of names is
+then ordered and slotted into a template which is then appended to the
+output.  Once the program has reached the last name in the list, it
+outputs the results.
 
-The second point of automation is the numerical pedigrees, for 
-instance from LL's [Genelach h-Úa
+The second point of automation is the numerical components of
+pedigrees, for instance from LL's [Genelach h-Úa
 Lomthuile](https://celt.ucc.ie/published/G800011F/text002.html):
 
 ```
 Trí mc la Cummine .i. Laidcnén. Conandil. Suibne.
 ```
 
 The utility can recognise up to the number ten and produce snippets of
-RDF for the curator to integrate into the database.
+RDF for the curator to integrate into the database.  For instance,
+using IrishGen's
+[numbered.pl](https://github.com/cyocum/irish-gen/blob/master/utils/numbered.pl):
+
+```bash
+➜  irish-gen git:(master) ✗ perl -CA utils/numbered.pl "Trí mc la Cummine .i. Laidcnén. Conandil. Suibne."
+<#laCummine>
+ a foaf:Person;
+irishRel:genName "la Cummine";
+irishRel:numChild 3.
+
+<#Laidcnén>
+ a foaf:Person;
+irishRel:nomName "Laidcnén";
+rel:childOf <#laCummine>.
+
+<#Conandil>
+ a foaf:Person;
+irishRel:nomName "Conandil";
+rel:childOf <#laCummine>.
+
+<#Suibne>
+ a foaf:Person;
+irishRel:nomName "Suibne";
+rel:childOf <#laCummine>.
+```
+
+This utility works by having the first ten numerals in [Old
+Irish](http://www.worldcat.org/oclc/899506131) directly encoded within
+it.  It then splits on the ".i." to separate the parent from the
+children.  Subsequently, it splits on the "mc" (or a few other
+variants that have been encountered in the sources).  The numeral
+which comes before the "m" becomes the `numChild`, the numeral is
+matched to the Old Irish word and translated into a more familiar
+Arabic numeral, while the rest of the parent indicator becomes the
+name.  As one will notice with `irishRel:genName "la Cummine"` this
+causes the incorrect name to be generated so human intervention is
+necessary to clean up the output before integrating it into the
+dataset.  The rest is slotting the children and parent into their RDF
+templates then outputting the result.
 
```

Comment:
One aspect of using this utility to consider is the need to then check within the file to which the output has been added to make sure the URLs in the output are unique within the file. This is another human intervention. Although I imagine we'll have one or several blogposts on distinguishing URLs so maybe this could be covered then.